src/Documentation/doc/advanced/corporation/optimal-selling-price-market-ta2.md

```diff
@@ -0,0 +1,108 @@
+
+## Market price and markup limit
+
+Market price:
+
+- Material: `material.marketPrice`.
+- Product: `product.productionCost`. This value is based on `ProductMarketPriceMult`, input materials' `MarketPrice` and `Coefficient`.
+  - $n = {Number\ of\ input\ materials}$
+  - $ProductMarketPriceMult = 5$
+
+$$ProductMarketPrice = ProductMarketPriceMult\ast\sum_{i = 1}^{n}{MaterialMarketPrice_i\ast MaterialCoefficient_i}$$
+
+Markup limit:
+
+- Material:
+
+$$MaterialMarkupLimit = \frac{MaterialQuality}{MaterialMarkup}$$
+
+- Product:
+
+$$ProductMarkupLimit = \frac{Max(ProductEffectiveRating,0.001)}{ProductMarkup}$$
+
+## Maximize sales volume
+
+Define:
+
+$$ExpectedSalesVolume = \frac{ProducedUnits}{10}$$
+
+- This means we want to sell all produced units.
+
+In cycle's SALE state, game calculates `MaxSalesVolume` of material/product. If we set price too high, `MaxSalesVolume` is penalized. In order to maximize profit, we have to set the highest possible price while `MaxSalesVolume` is still equals to `ExpectedSalesVolume`. This is what Market-TA2 does.
+
+Calculation of material and product is pretty similar, so I'll call them "item" and use 1 formula.
+
+`MaxSalesVolume` is the product of 7 multipliers:
+
+- Item multiplier:
+  - Material: $ItemMultiplier = MaterialQuality + 0.001$
+  - Product: $ItemMultiplier = 0.5\ast(ProductEffectiveRating)^{0.65}$
+- Business factor:
+
+$$BusinessProduction = 1 + office.employeeProductionByJob["Business"]$$
+
+$${BusinessFactor = (BusinessProduction)}^{0.26} + \left( \frac{BusinessProduction}{1000} \right)$$
+
+- Advert factor:
+  - $AwarenessFactor = (Awareness + 1)^{IndustryAdvertisingFactor}$
+  - $PopularityFactor = (Popularity + 1)^{IndustryAdvertisingFactor}$
+
+$$RatioFactor = \begin{cases}Max(0.01,\frac{Popularity + 0.001}{Awareness}), & Awareness \neq 0 \newline 0.01, & Awareness = 0 \end{cases}$$
+
+$$AdvertFactor = (AwarenessFactor\ast PopularityFactor\ast RatioFactor)^{0.85}$$
+
+- Market factor:
+
+$$MarketFactor = Max\left( 0.1,\frac{Demand\ast(100 - Competition)}{100} \right)$$
+
+- Corporation's upgrade bonus: `SalesBots` bonus.
+- Division's research bonus: this is always 1. Currently there is not any research that increases the sales bonus.
+- `MarkupMultiplier`: initialize with 1.
+
+  - `SellingPrice` is the selling price that you set.
+  - With materials, if we set `SellingPrice` to 0, `MarkupMultiplier` is $10^{12}$ (check the formula below). Extremely high `MarkupMultiplier` means that we can sell all units, regardless of other factors. This is the fastest way to discard materials.
+  - If `(SellingPrice > MarketPrice + MarkupLimit)`:
+
+  $$MarkupMultiplier = \left( \frac{MarkupLimit}{SellingPrice - MarketPrice} \right)^{2}$$
+
+  - If item is material and `SellingPrice` is less than `MarketPrice`:
+
+$$MarkupMultiplier = \begin{cases}\frac{MarketPrice}{SellingPrice}, & SellingPrice > 0 \land SellingPrice < MarketPrice \newline 10^{12}, & SellingPrice \leq 0 \end{cases} $$
```

Comment:
Optional: Given the audience, I think `\&\&` would work better than `\land` for "and".

Reply:
GitHub has bug that renders the ampersand incorrectly. There are workarounds for that bug, but those workarounds are rendered incorrectly in the in-game documentation.
Ref: https://github.com/orgs/community/discussions/46788

Reply:
GitHub has a bug that renders the ampersand incorrectly. There are workarounds for that bug, but those workarounds are rendered incorrectly in the in-game documentation.
Ref: https://github.com/orgs/community/discussions/46788